### PR TITLE
tests: add e2e hbone test

### PR DESCRIPTION
**Pull request by howardjohn**
Previously we had a TCP one, this adds HBONE. The challenge here is TLS; we needed a way to use a mock CA.

* Use `dyn CertificateProvider` in proxy to avoid generic proliferation
* Make the test certificate generation generate real certifications -- the old ones were invalid
* Regen the test certs (and add script) so that we have the ca key and cert as well

### CODE REVIEW

File: src/test_helpers/app.rs [hunk 1]
```diff
@@ -35,7 +36,8 @@ where
 {
     initialize_telemetry();
 
-    let app = app::build(cfg).await.unwrap();
```
Comment (stevenctl) [inline src/test_helpers/app.rs:38]: Not needed for this PR but I think we should try to split up `app::build` (maybe needs an `AppBuilder` struct) so we can easily override things that can't be set via config. (like manually creating a MockCaClient with bad certs, or custom logic to give bad certs for only one ID). 

`app::new` would be where we set our defaults.

File: tests/proxy.rs [hunk 2]
```diff
@@ -101,6 +96,16 @@ async fn test_request() {
     .await;
 }
 
+#[tokio::test]
+async fn test_hbone_request() {
```
Comment (stevenctl) [inline tests/proxy.rs:100]: Should be ~trivial to add some negative tests for expired cert

**Review by stevenctl — APPROVED** [review]
Nice! Added DNM so you can merge when ready.

File: src/test_helpers/app.rs (hunk 1 above)
Comment (howardjohn) [reply src/test_helpers/app.rs:38]: I was making an AppBuilder at first but got annoyed so went with this for now :-)
 but I agree

**Pull request merged**